src/MongoDB.Driver.Core/Core/Connections/ClientDocumentHelper.cs

```diff
@@ -59,11 +59,22 @@ internal static BsonDocument CreateDriverDocument()
 
         internal static BsonDocument CreateDriverDocument(string driverVersion)
         {
+            var driverName = "mongo-csharp-driver";
+            if (IsLegacyLoaded())
+            {
+                driverName = $"{driverName}|legacy";
```

Comment:
I think it would look nicer to have a "-" between the original driver name and "legacy":

    mongo-csharp-driver-legacy

Reply:
this is how it's suggested in the ticket description by Jeff:

>So for users of the the MongoDB.Driver module, the name can stay the same. But for users of the MongoDB.Driver.Legacy module, it should be "mongo-csharp-driver|legacy".

I will change it if everyone agrees

cc: @jyemin

Reply:
I didn't realize that the ticket was that prescriptive.

The vertical bar looks weird, but I can go along with whatever @jyemin  wants it to be.

Note that we can't really change our minds later once we start collecting metadata.

Reply:
It's what the java driver has been using for ages.  Better to be consistent about it.